connectors/src/connectors/intercom/lib/intercom_api.ts

```diff
@@ -290,11 +330,139 @@ export async function fetchIntercomTeams(
  * Return the detail of a Team.
  */
 export async function fetchIntercomTeam(
-  intercomClient: Client,
+  nangoConnectionId: string,
   teamId: string
-): Promise<TeamObject | null> {
-  const teamsResponse = await intercomClient.teams.find({
-    id: teamId,
+): Promise<IntercomTeamType | null> {
+  if (!NANGO_INTERCOM_CONNECTOR_ID) {
+    throw new Error("NANGO_NOTION_CONNECTOR_ID not set");
+  }
+
+  const accessToken = await getAccessTokenFromNango({
+    connectionId: nangoConnectionId,
+    integrationId: NANGO_INTERCOM_CONNECTOR_ID,
+    useCache: true,
+  });
+
+  const resp = await fetch(`https://api.intercom.io/teams/${teamId}`, {
+    method: "GET",
+    headers: {
+      "Intercom-Version": "2.10",
+      Authorization: `Bearer ${accessToken}`,
+      "Content-Type": "application/json",
+    },
+  });
+
+  const response = await resp.json();
+  return response;
+}
+
+/**
+ * Return the paginated list of Conversation for a given Team.
+ * Filtered on the last 3 months and closed Conversations.
+ */
+export async function fetchIntercomConversationsForTeamId({
+  nangoConnectionId,
+  teamId,
+  cursor = null,
+  pageSize = 20,
+}: {
+  nangoConnectionId: string;
+  teamId: string;
+  cursor: string | null;
+  pageSize?: number;
+}): Promise<{
+  conversations: IntercomConversationType[];
+  pages: {
+    next?: {
+      page: number;
+      starting_after: string;
+    };
+  };
+}> {
+  if (!NANGO_INTERCOM_CONNECTOR_ID) {
+    throw new Error("NANGO_NOTION_CONNECTOR_ID not set");
+  }
+
+  const accessToken = await getAccessTokenFromNango({
+    connectionId: nangoConnectionId,
+    integrationId: NANGO_INTERCOM_CONNECTOR_ID,
+    useCache: true,
+  });
+
+  const ninetyDaysAgo = new Date(Date.now() - 90 * 24 * 60 * 60 * 1000);
```

Comment:
Should we use that as a parameter (as we likely will want to store that window as configuration of the connector)?

Reply:
I can def rename the model `IntercomWorkspace` to `IntercomConfiguration` and store this as a parameter inside!